src/github.com/getlantern/flashlight/localdiscovery/localdiscovery.go

```diff
@@ -0,0 +1,79 @@
+// service for discovering Lantern instances in the local network
+package localdiscovery
+
+import (
+	"github.com/getlantern/flashlight/ui"
+	"github.com/getlantern/golog"
+	"github.com/getlantern/multicast"
+	"sync"
+	"time"
+)
+
+const (
+	messageType  = `LocalDiscovery`
+	updatePeriod = 10
+)
+
+var (
+	log           = golog.LoggerFor("flashlight.localdiscovery")
+	service       *ui.Service
+	mc            *multicast.Multicast
+	lastPeers     []multicast.PeerInfo
+	peersMutex    sync.Mutex
+)
+
+func Start(portToAdvertise string) {
+	if service == nil {
+		var err error
+		service, err = ui.Register(messageType, nil, func(write func(interface{}) error) error {
+			return write(buildPeersList())
+		})
+		if err != nil {
+			log.Errorf("Unable to register Local Discovery service: %q", err)
+			return
+		}
+	}
```

Comment:
This nil check is to guard against Start being called multiple times, correct @uaalto? If so maybe sync.Once is a better choice, as in:

```go
once.Do(start(portToAdvertise))
```

with ```start``` as a new private method?